common/src/main/java/com/wynntils/features/user/ChatCoordinatesFeature.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.objects.Location;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.mc.utils.LocationUtils;
+import com.wynntils.wynn.event.ChatMessageReceivedEvent;
+import com.wynntils.wynn.model.CompassModel;
+import com.wynntils.wynn.utils.WynnUtils;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.Style;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ChatCoordinatesFeature extends UserFeature {
+    @Override
+    public List<Class<? extends Model>> getModelDependencies() {
+        return List.of(CompassModel.class);
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onChatReceived(ChatMessageReceivedEvent e) {
+        if (!WynnUtils.onWorld()) return;
+
+        Component message = e.getMessage();
+
+        e.setMessage(insertCoordinateComponents(message));
+    }
+
+    public static Component insertCoordinateComponents(Component message) {
```

Comment:
Yes, this is taken from ChatItemModel. I don't think we need to generalize this, as we don't really have any other use case for this (and it would be weird)

Reply:
I'm starting to think that maybe the ChatModel should have some kind of "transformers" registered on it. And "redirectors", that capture chats that should be redirected to the notification manager. Just a hand-wavy idea, not requesting that you change anything here.

Reply:
Perhaps, although this case is rare enough not to add it for now